dimod/binary_quadratic_model.py

```diff
@@ -1546,24 +1532,28 @@ def to_json(self, fp=None):
              "variable_labels": ["a", "b"], "variable_type": "SPIN",
              "version": {"bqm_schema": "1.0.0", "dimod": "0.6.3"}}
 
+            This is an example of writing a binary quadratic model to a JSON-format file.
+
+            >>> import dimod
+            >>> bqm = dimod.BinaryQuadraticModel({'a': -1.0, 'b': 1.0}, {('a', 'b'): -1.0}, 0.0, dimod.SPIN)
+            >>> with open('tmp.txt', 'w') as file:  # doctest: +SKIP
+            ...     file.write(bqm.to_json())
+
         """
         import json
         from dimod.io.json import DimodEncoder
 
-        if fp is None:
-            return json.dumps(self, cls=DimodEncoder, sort_keys=True)
-        else:
-            return json.dump(self, fp, cls=DimodEncoder, sort_keys=True)
+        return json.dumps(self, cls=DimodEncoder, sort_keys=True)
 
     @classmethod
     def from_json(cls, obj):
         """Deserialize a binary quadratic model from a JSON encoding.
 
         Args:
             obj: (str/file):
-                Either a string or a  `.read()`-supporting `file object`_
+                Either a string or a `.read()`-supporting `file object`_
```

Comment:
Not sure about this.. On one hand, it's good that `BQM.from_json(bqm.to_json())` works as expected, but on the other, it's inconsistent with `to_json` interface.